polynote-spark/src/test/scala/polynote/kernel/MockKernel.scala

```diff
@@ -0,0 +1,105 @@
+package polynote.kernel
+
+import java.nio.ByteBuffer
+
+import cats.effect.IO
+import cats.syntax.traverse._
+import cats.instances.list._
+import fs2.Stream
+import polynote.config.PolynoteConfig
+import polynote.kernel.util.Publish
+import polynote.messages.{ByteVector32, CellID, CellResult, HandleType, Notebook, NotebookUpdate, Streaming}
+import polynote.runtime._
+import DataEncoder.StructDataEncoder
+import polynote.server.KernelFactory
+import scodec.bits.ByteVector
+
+import scala.reflect.runtime.universe.NoType
+
+object CellIdSyntax {
+  class CellIdSyntax(val i: Int) extends AnyVal {
+    def cell: CellID = i.toShort
+  }
+  // implicit class isn't recognized for some reason?
+  implicit def intToCellId(i: Int): CellIdSyntax = new CellIdSyntax(i)
+}
+
+import CellIdSyntax._
+
+class MockKernelFactory(val kernel: MockKernel) extends KernelFactory[IO] {
+  def launchKernel(getNotebook: () => IO[Notebook], statusUpdates: Publish[IO, KernelStatusUpdate], config: PolynoteConfig): IO[KernelAPI[IO]] =
+    IO.pure(kernel)
+}
+
+
+// TODO: add more utility to this - maybe record/check API
+class MockKernel(@volatile private var notebook: Notebook) extends KernelAPI[IO] {
+  def init: IO[Unit] = IO.unit
+  def shutdown(): IO[Unit] = IO.unit
+
+  def startInterpreterFor(id: CellID): IO[fs2.Stream[IO, Result]] = IO.pure {
```

Comment:
`Stream` would work fine here, no need for `fs2.`

Reply:
Good point... had IntelliJ implement the trait before I imported `fs2.Stream`. Will fix

Reply:
Fixed